### ISSUE TITLE
Regent: bug vanishes on adding printf statements

### ISSUE THREAD

**Issue by rupanshusoi** [StanfordLegion/legion#727]
I have a function that returns an array of 4 doubles. Surprisingly only the last double is being miscalculated. You can look at the code here: https://filebin.net/kz5e4hfikrme05w2
You can see that the output has changed by just adding two printf statements.

I figured out that after adding the 3rd element to the array (line 34), changing `temp1` in line 38 is causing `temp2` to also change. Substituting the value of `temp1` in the expression for `temp2` in lines 36, 37
```
temp1 = (7 * pr_by_rho) + u_sqr
temp2 = 0.5 * ut * temp1  * A1neg
```
like so
```
temp1 = (7 * pr_by_rho) + u_sqr
temp2 = 0.5 * ut * ((7 * pr_by_rho) + u_sqr)  * A1neg
```
also fixes the bug.

**Comment by elliottslaughter** [comment]
Confirmed. Diffing the output of `-fpretty 1` it looks like some statements are being rearranged. I'm currently thinking this must be in the normalizer because running with `-fflow 0` doesn't change anything, and I'm not sure what else is capable of doing this.

```
$ ./regent.py test_bug_727.rg -fpretty 1 > test_bug_727_pretty.txt
$ ./regent.py test_bug_727_fixed.rg -fpretty 1 > test_bug_727_fixed_pretty.txt
$ diff -u test_bug_727_pretty.txt test_bug_727_fixed_pretty.txt 
--- test_bug_727_pretty.txt	2020-01-28 16:08:47.572125726 -0800
+++ test_bug_727_fixed_pretty.txt	2020-01-28 16:08:55.132075054 -0800
@@ -30,10 +30,12 @@
   $temp1 = (($ut*$A1neg)-$B1)
   $G[2] = (($rho*$temp1)*$temp2)
   $temp1 = ((7*$pr_by_rho)+$u_sqr)
-  $temp1 = ((6*$pr_by_rho)+$u_sqr)
   $temp2 = (((0.5*$ut)*$temp1)*$A1neg)
+  printf("temp2 = %0.15lf\n", $temp2)
+  $temp1 = ((6*$pr_by_rho)+$u_sqr)
   var $temp3 : double = ((0.5*$B1)*$temp1)
   $temp1 = (($ut*$A1neg)-$B1)
+  printf("temp2 = %0.15lf\n", $temp2)
   var $temp4 : double = ((((0.5*$rho)*$un)*$B2)*$temp1)
   $G[3] = ((($rho*$A2pos)*($temp2-$temp3))+$temp4)
   return $G
$ diff -u test_bug_727.rg test_bug_727_fixed.rg 
--- test_bug_727.rg	2020-01-28 16:06:36.553002521 -0800
+++ test_bug_727_fixed.rg	2020-01-28 16:07:39.420582147 -0800
@@ -35,9 +35,11 @@
 
         temp1 = (7 * pr_by_rho) + u_sqr
         temp2 = 0.5 * ut * temp1  * A1neg
+	C.printf("temp2 = %0.15lf\n", temp2)
         temp1 = (6 * pr_by_rho) + u_sqr
         var temp3 = 0.5 * B1 * temp1
         temp1 = ut * A1neg - B1
+	C.printf("temp2 = %0.15lf\n", temp2)
         var temp4 = 0.5 * rho * un * B2 * temp1
 	G[3] = rho * A2pos * (temp2 - temp3) + temp4
````

**Comment by elliottslaughter** [comment]
Ok, @magnatelee convinced me this is an RDIR bug. Attaching a smaller reproducer:

```
import "regent"

task f()
  var x = 0
  var y = 0
  x = 123
  y = 2 * x
  x = 456
  return y
end

task main()
  regentlib.c.printf("%d\n", f())
end
regentlib.start(main)
```

Output now varies with the addition/removal of `-fflow 0`.

**Comment by elliottslaughter** [comment]
Ok, this should be fixed in fbc149e0f5b5a5f0b3ab4cab1686c2b7d6405637, which has been pushed to `master`. Note that you will also need to update RDIR (`git submodule update --init`).
